src/uwsgicache.py

```diff
@@ -52,9 +52,10 @@ def get(self, key, default=None, version=None):
             return pickle.loads(val)
 
         def _set(self, full_key, value, timeout):
-            if timeout is True:
+            if timeout is True or timeout == DEFAULT_TIMEOUT:
                 uwsgi_timeout = self.default_timeout
-            elif timeout is None or timeout is False:
+
```

Comment:
It's there so the values from `settings.py` can be Django values. If you keep it as elif, the values defined in `settings.py` needs to be uwsgi specific values and not Django values, like `-1` for for timeout, which has nothing to do with Django at all. I think this fits [Principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)